core/src/setup/Kokkos_Setup_SYCL.hpp

```diff
@@ -62,4 +62,12 @@ using sycl_host_ptr = sycl::host_ptr<T>;
 #endif
 }  // namespace Kokkos::Impl
 
+#ifdef KOKKOS_ENABLE_SYCL_RELOCATABLE_DEVICE_CODE
+#define KOKKOS_IMPL_EXTERNAL_FUNCTION SYCL_EXTERNAL
+#else
+#define KOKKOS_IMPL_EXTERNAL_FUNCTION   \
+  @"KOKKOS_EXTERNAL_FUNCTION requires " \
+   "Kokkos_ENABLE_SYCL_RELOCATABLE_DEVICE_CODE=ON"
```

Comment:
Please print sample error message when this is used

Reply:
```
/home/darndt/kokkos/example/build_cmake_rdc/functor.cpp:19:1: error: expected unqualified-id
   19 | KOKKOS_EXTERNAL_FUNCTION void count_even(const long i, long& lcount) {
      | ^
/home/darndt/kokkos-install/include/Kokkos_Macros.hpp:383:34: note: expanded from macro 'KOKKOS_EXTERNAL_FUNCTION'
  383 | #define KOKKOS_EXTERNAL_FUNCTION KOKKOS_IMPL_EXTERNAL_FUNCTION
      |                                  ^
/home/darndt/kokkos-install/include/setup/Kokkos_Setup_SYCL.hpp:69:3: note: expanded from macro 'KOKKOS_IMPL_EXTERNAL_FUNCTION'
   69 |   @"KOKKOS_EXTERNAL_FUNCTION requires " \
      |   ^
1 error generated.
```